src/main/java/org/testng/Assert.java

```diff
@@ -369,16 +369,13 @@ public static void assertEquals(float[] actual, float[] expected, String message
     if (checkRefEqualityAndLength(actual, expected, message)) return;
 
     for (int i = 0; i < expected.length; i++) {
-      if (expected[i] != actual[i]) {
-        fail(
-            String.format(
-                ARRAY_MISMATCH_TEMPLATE,
-                i,
-                Float.toString(expected[i]),
-                Float.toString(actual[i]),
-                message));
+      assertEquals(actual[i], expected[i], String.format(
+                                              ARRAY_MISMATCH_TEMPLATE,
```

Comment:
Not sure what the logic for this formatting is, it should just be indented one or possibly two from the previous line.